go-runtime/ftl/ftltest/ftltest.go

```diff
@@ -168,6 +171,64 @@ func WithSecret[T ftl.SecretType](secret ftl.SecretValue[T], value T) Option {
 	}
 }
 
+// WithDatabase sets up a database for testing by appending "_test" to the DSN and emptying all tables
+//
+// To be used when setting up a context for a test:
+// ctx := ftltest.Context(
+//
+//	ftltest.WithDatabase(db),
+//	... other options
+//
+// )
+func WithDatabase(dbHandle ftl.Database) Option {
+	return func(ctx context.Context, state *OptionsState) error {
+		envarName := fmt.Sprintf("FTL_POSTGRES_DSN_%s_%s", strings.ToUpper(ftl.Module()), strings.ToUpper(dbHandle.Name))
+		originalDSN, ok := os.LookupEnv(envarName)
+		if !ok {
+			return fmt.Errorf("missing DSN for database %s: expected to find it at the environment variable %s", dbHandle.Name, envarName)
+		}
+
+		// convert DSN by appending "_test" to table name
+		// postgres DSN format: postgresql://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]
+		// source: https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING
+		dsnURL, err := url.Parse(originalDSN)
+		if err != nil {
+			return fmt.Errorf("could not parse DSN: %w", err)
+		}
+		if dsnURL.Path == "" {
+			return fmt.Errorf("DSN for %s must include table name: %s", dbHandle.Name, originalDSN)
+		}
+		dsnURL.Path += "_test"
+		dsn := dsnURL.String()
+
+		// connect to db and clear out the contents of each table
+		sqlDB, err := sql.Open("pgx", dsn)
+		if err != nil {
+			return fmt.Errorf("could not create database %q with DSN %q: %w", dbHandle.Name, dsn, err)
+		}
+		_, err = sqlDB.ExecContext(ctx, `DO $$ 
+		DECLARE 
+		   table_name text; 
+		BEGIN 
+		   FOR table_name IN (SELECT tablename FROM pg_tables WHERE schemaname = 'public') 
+		   LOOP 
+			  EXECUTE 'DELETE FROM ' || table_name; 
+		   END LOOP; 
+		END $$;`)
+		if err != nil {
+			return fmt.Errorf("could not clear tables in database %q: %w", dbHandle.Name, err)
+		}
```

Comment:
Nice!

We probably need to be slightly smarter about this, in that the table that tracks migrations will also be wiped. But that can probably wait until we have first-class support for migrations.